openhands/agenthub/codeact_agent/codeact_agent.py

```diff
@@ -169,14 +169,20 @@ def get_action_message(
             logger.debug(
                 f'Tool calls type: {type(assistant_msg.tool_calls)}, value: {assistant_msg.tool_calls}'
             )
-            pending_tool_call_action_messages[llm_response.id] = Message(
+            # Create message with token counts from Usage data
+            message = Message(
                 role=assistant_msg.role,
                 # tool call content SHOULD BE a string
                 content=[TextContent(text=assistant_msg.content or '')]
                 if assistant_msg.content is not None
                 else [],
                 tool_calls=assistant_msg.tool_calls,
+                event_id=action.id,
```

Comment:
This is incorrect for multiple tool calls. We're trying to make sure there "shouldn't be"(tm) multiple tool calls, but there might be.
For non-native function calling LLMs, this is even more clear.